### PR TITLE
Workaround gcc-4.8 bug for TensorFlow.

### PR DESCRIPTION
**Pull request by coryan**
While we do not support gcc-4.8, TensorFlow does, and it was easy
to workaround the compiler bugs this time.  We may have to revisit
the decision to drop gcc-4.8 support altogether.  For now, just
make the code work for the TensorFlow folks.

### CODE REVIEW

File: google/cloud/bigtable/internal/rowreaderiterator.h [hunk 1]
```diff
@@ -54,7 +54,6 @@ class RowReaderIterator : public std::iterator<std::input_iterator_tag, Row> {
 
   Row const& operator*() const& { return *row_; }
   Row& operator*() & { return *row_; }
-  Row const&& operator*() const&& { return *std::move(row_); }
```
Comment (mbrukman) [inline google/cloud/bigtable/internal/rowreaderiterator.h]: Do you want to make this a conditional thing, i.e., if GCC && GCC version >= 4.9 or similar? Or is this method really not necessary?

Comment (coryan) [reply google/cloud/bigtable/internal/rowreaderiterator.h]: > Do you want to make this a conditional thing, i.e., if GCC && GCC version >= 4.9 or similar?

I think that makes more sense, now that you mention it.

> Or is this method really not necessary?

Well, the standard defines it for a reason (it is one of the canonical operators for an iterator), but they happen in very rare circumstances.  None of the code we have uses it for example.

**Comment by coryan** [comment]
PTAL. Also created #798 to discuss the decision to drop gcc-4.8.

**Comment by coryan** [comment]
Oh, and the ugly formatting is all clang-format's fault.

**Pull request merged**